fix-file-permissions.sh

```
@@ -0,0 +1,38 @@
+find "${public_path}/" -type d ! -perm ug=rwx,o=rx -print0 | xargs -r -0 -L20 chmod ug=rwx,o=rx
+find "${public_path}/" ! -user ${www_id} -print0 | xargs -r -0 -L20 chown ${www_id}
+find "${public_path}/" ! -group ${www_id} -print0 | xargs -r -0 -L20 chgrp ${www_id}
+
+# PRIVATE: Privately owned and accessible by PHP process
+find "${private_path}/" -type f ! -perm ug=rw,o= -print0 | xargs -r -0 -L20 chmod ug=rw,o=
+find "${private_path}/" -type d ! -perm ug=rwx,o= -print0 | xargs -r -0 -L20 chmod ug=rwx,o=
+find "${private_path}/" ! -user ${www_id} -print0 | xargs -r -0 -L20 chown ${www_id}
+find "${private_path}/" ! -group ${www_id} -print0 | xargs -r -0 -L20 chgrp ${www_id}
+
+# Ensure public thumbnails are not writable (prevent deletion):
+find ${public_path}/styles/pece_thumbnail -type f ! -perm ugo=r -print0 | xargs -r -0 -L20 chmod ugo=r
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)